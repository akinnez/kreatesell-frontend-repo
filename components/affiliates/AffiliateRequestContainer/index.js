import Link from "next/link";
import { useRouter } from "next/router";
import { useSelector } from "react-redux";
import { Card, Tabs } from "antd";
import { MdOutlineRemoveRedEye } from "react-icons/md";
import BackButton from "components/BackButton";
import styles from "./index.module.scss";

const AffiliateRequestContainer = ({
  children,
  productTypeId,
  productName,
}) => {
  const router = useRouter();
  const { productTypes } = useSelector(state => state.product);
  const productType = productTypes.find(type => type.id === +productTypeId);

  return (
    <>
      <header className={styles.header}>
        <span>
          <BackButton />
        </span>
        <h1>
          {productName} - <span>{productType?.product_type_name}</span>
        </h1>
        <Link href={`/account/affiliate/preview/product/${router.query.pId}`}>
          <a>
            <MdOutlineRemoveRedEye /> View Product Page
          </a>
        </Link>
      </header>
      <div className={styles.card__container}>
        <Tabs className={styles.tabs} defaultActiveKey="1" centered>
          {children}
        </Tabs>
      </div>
    </>
  );
};

export default AffiliateRequestContainer;
