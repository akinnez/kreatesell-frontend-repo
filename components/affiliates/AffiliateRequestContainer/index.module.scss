@use 'public/css/colors';
@use 'public/css/mixins';
@use 'public/css/screens';

.header {
	@include mixins.flex(column, space-between, flex-start);
	row-gap: 10px;
	margin-bottom: 25px;

	> span {
		margin-bottom: 10px;
	}

	h1 {
		margin-bottom: 0;
		font: {
			size: 18px;
			weight: bold;
		}
		color: colors.$black-100;
		word-break: break-word;

		span {
			color: colors.$gray-200;
		}
	}

	a {
		min-width: 182px;
		display: inline-flex;
		padding: 10px 20px;
		background-color: transparent;
		border-radius: 8px;
		border: 1px solid colors.$base-blue;
		color: colors.$base-blue;
		font-size: 12px;
		font-weight: bold;

		svg {
			margin-right: 10px;
			font-size: 18px;
		}
	}
}

.card__container {
	margin-bottom: 275px;
}

.tabs {
	div[class='ant-tabs-nav'] div[class^='ant-tabs-ink-bar']:last-child {
		height: 4px;
	}

	div[class='ant-tabs-tab-btn'] {
		color: colors.$base-gray;
		font: {
			weight: 500;
			size: 16px;
		}
	}

	div[class$='ant-tabs-tab-active'] div[class='ant-tabs-tab-btn'] {
		color: colors.$base-blue;
	}
}

@include screens.screen('x-large') {
	.tabs {
		div[class='ant-tabs-nav'] {
			margin-bottom: 16px;
			background-color: colors.$base-white;
			border-radius: 8px;
			padding-left: 7px;
			padding-right: 7px;

			&::before {
				border: none;
			}

			div[class='ant-tabs-nav-wrap']:first-child {
				justify-content: flex-start;
			}
		}

		div[class='ant-tabs-tab'] + div[class='ant-tabs-tab'],
		div[class='ant-tabs-tab ant-tabs-tab-active']
			+ div[class='ant-tabs-tab'],
		div[class='ant-tabs-tab']
			+ div[class='ant-tabs-tab ant-tabs-tab-active'] {
			margin-left: 5px;
		}

		div[class='ant-tabs-tab-btn'] {
			padding: 8px;
			border-radius: 8px;
			background-color: colors.$white-200;
		}

		div[class$='ant-tabs-tab-active'] div[class='ant-tabs-tab-btn'] {
			background-color: colors.$blue-100;
		}
	}
}

@include screens.minScreen('medium') {
	.header {
		@include mixins.flex(row, space-between, center);
		flex-wrap: wrap;
		column-gap: 20px;

		> span {
			flex-basis: 100%;
		}

		h1 {
			flex-basis: calc(100% - 210px);
		}
	}
}

@include screens.minScreen('x-large') {
	.header {
		flex-wrap: nowrap;

		> span {
			flex-basis: auto;
			margin-bottom: 0;
		}

		h1 {
			text-align: center;
			flex-basis: auto;
		}
	}

	.card__container {
		padding: 60px 75px;
		background-color: colors.$base-white;
		border-radius: 8px;
	}

	.tabs > div[class='ant-tabs-nav'] {
		margin-bottom: 30px;

		&::before {
			border-bottom-color: colors.$gray-700;
		}
	}
}
