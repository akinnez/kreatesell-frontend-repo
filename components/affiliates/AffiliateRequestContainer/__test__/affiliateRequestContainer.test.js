import AffiliateRequestContainer from "..";
import renderer from "react-test-renderer";

it("renders a snapshot of <AffiliateRequestContainer />", () => {
  const container = renderer
    .create(
      <AffiliateRequestContainer>mock Children</AffiliateRequestContainer>
    )
    .toJSON();
  expect(container).toMatchSnapshot();
});
