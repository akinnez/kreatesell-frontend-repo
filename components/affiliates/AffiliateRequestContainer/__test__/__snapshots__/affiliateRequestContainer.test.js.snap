// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <AffiliateRequestContainer /> 1`] = `
Array [
  <div
    className="product__name"
  >
    <h1
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      The Land Of Hope - 
      <span
        className="ant-typography ant-typography-secondary"
        onClick={null}
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        DIGITAL DOWNLOAD
      </span>
    </h1>
  </div>,
  <div
    className="ant-card ant-card-bordered card__container"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-tabs ant-tabs-top ant-tabs-centered tabs"
      >
        <div
          className="ant-tabs-nav"
          onKeyDown={[Function]}
          role="tablist"
        >
          <div
            className="ant-tabs-nav-wrap"
          >
            <div
              className="ant-tabs-nav-list"
              style={
                Object {
                  "transform": "translate(0px, 0px)",
                  "transition": undefined,
                }
              }
            >
              <div
                className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
          <div
            className="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
          >
            <button
              aria-controls="null-more-popup"
              aria-expanded={false}
              aria-haspopup="listbox"
              aria-hidden="true"
              className="ant-tabs-nav-more"
              id="null-more"
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "marginLeft": undefined,
                  "order": 1,
                  "visibility": "hidden",
                }
              }
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="ellipsis"
                className="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          className="ant-tabs-content-holder"
        >
          <div
            className="ant-tabs-content ant-tabs-content-top"
            style={null}
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
